src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1643,12 +1644,12 @@ internal static object CopyDateTimeOffset(object obj, ICopyContext context)
 
         internal static void SerializeType(object obj, ISerializationContext context, Type expected)
         {
-            context.StreamWriter.Write(((Type)obj).OrleansTypeKeyString());
+            context.StreamWriter.Write(RuntimeTypeNameFormatter.Format((Type)obj));
         }
 
         internal static object DeserializeType(Type expected, IDeserializationContext context)
         {
-            return context.SerializationManager.ResolveTypeName(context.StreamReader.ReadString());
+            return Type.GetType(context.StreamReader.ReadString());
```

Comment:
I assume you chose to use `Type.GetType` as opposed to `TypeUtils.ResolveType` since the type is always qualified with the assembly name, hence assembly loading is not an issue?